src/lib/utils.h

```diff
@@ -56,6 +59,24 @@
         }                                                      \
     } while (/* CONSTCOND */ 0)
 
+/*
+ * @params
+ * array:       array of the structures to lookup
+ * id_field     name of the field to compare against
+ * ret_field    filed to return
+ * lookup_value lookup value
+ * ret          return value
+ */
+#define ARRAY_LOOKUP_BY_ID(array, id_field, ret_field, lookup_value, ret) \
```

Comment:
I think you were going to change this to `ARRAY_SEARCH_BY_ID` (https://github.com/riboseinc/rnp/pull/537#discussion_r151877028).

There is some nuance here and search is the more accurate descriptor.

Reply:
Thanks. I've decided that I prefer ``lookup`` as this is what this macro does - it retrieves a record from a table corresponding to lookup_value. it doesn't search whole array

Reply:
>  it doesn't search whole array

Well, surely it does search the whole array when it doesn't find what it's looking for.

Anyways, one distinction is that `lookup` implies that you are retrieving something that *definitely* exists, whereas `search` implies you are trying to find something that *may* exist. In our case, it's the latter - we may not find what we are looking for.

http://www.thefreedictionary.com/look+up
> To search for and find, as in a reference book.

https://english.stackexchange.com/a/165859

But I can certainly see some points in favor of either terms, and I'm sure developers are going to read the macro contents anyways.

Reply:
So my way of thinking is - when "calling" this macro you expect that value exist (otherwise it's an error condition), you have a method for locating candidates (lookup_value) and what's most important - you expect that there is only one result.

Reply:
Seems logical to me, let's just leave the name.